engine/conditions.go

```diff
@@ -0,0 +1,31 @@
+package engine
+
+type InnerConndtionResponse struct {
+	Passed bool
+	// For each time the condition is hit, add all of the context.
+	// keys here, will be used in the message.
+	ConditionHitContext []map[string]string
+}
+
+type InnerCondition interface {
+	Evaluate() (InnerConndtionResponse, error)
+}
+
+type Condition struct {
+	// Optional When clause
+	When *Conditional `json:"when,omitempty"`
+
+	InnerCondition `json:"innerCondition,omitempty"`
+}
+
+type Conditional struct {
```

Comment:
Not sure I follow the separation between logical conditions and the innercondition, could or/and not be implemented as InnerConditions whose `Evaluate` method calls the `Evaluate` methods of the subconditions?